modin/core/dataframe/algebra/tree_reduce.py

```diff
@@ -20,7 +20,7 @@ class TreeReduce(Operator):
     """Builder class for TreeReduce operator."""
 
     @classmethod
-    def register(cls, map_function, reduce_function=None, axis=None):
+    def register(cls, map_function, reduce_function=None, axis=None, dtypes=None):
```

Comment:
Maybe name it `compute_dtypes` to clarify it's a callable?

Reply:
Agree, changed